File: src/1342-Number-of-Steps-to-Reduce-Number-to-Zero/number_of_steps_to_reduce_number_to_zero.py
class Solution:
    def numberOfSteps(self, num: int) -> int:
        steps: int = 0
        while num > 0:
            if num & 1 == 0:
                num = num >> 1
            else:
                num -= 1
            steps += 1
        return steps
 

if __name__ == "__main__":
    num: int = 14
    solution = Solution()
    print(solution.numberOfSteps(num))

File: src/1342-Number-of-Steps-to-Reduce-Number-to-Zero/number_of_steps_to_reduce_number_to_zero.cpp
#include <iostream>

class Solution {
public:
    int numberOfSteps(int num) {
        int steps = 0;
        while (num > 0) {
            if ((num & 1) == 0) {
                num >>= 1;
            }
            else {
                num--;
            }
            steps++;
        }
        return steps;
    }
};

int main() {
    int num = 14;
    Solution solution;
    int result = solution.numberOfSteps(num);

    std::cout << result;
}